src/components/TodoItems.tsx

```diff
@@ -0,0 +1,198 @@
+          .then(todoValue => {
+            setTodos(prevTodos =>
+              prevTodos.map(currentTodo =>
+                currentTodo.id === todo.id ? todoValue : currentTodo,
+              ),
+            );
+            setIsEditing(false);
+            focusInput();
+            setQueryEditing(queryEditing.trim());
+          })
+          .catch(() => {
+            handleError(ErrorList.UpdateTodo);
+          })
+          .finally(() => {
+            setIsLoading(false);
+          });
+      }
+    } else {
+      setIsLoading(true);
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(todos.filter(item => item.id !== todo.id));
+          setIsEditing(false);
+          focusInput();
+        })
+        .catch(() => {
+          handleError(ErrorList.DeleteTodo);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  };
+
+  const handleClickKeybord = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      focusInput();
+      setIsEditing(false);
+      setQueryEditing(todo.title);
+    }
+
+    if (event.key === 'Enter') {
+      saveInputValue();
+    }
+  };
+
+  const handleQueryEditChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQueryEditing(event.target.value);
+  };
+
+  const handleBlurInput = () => {
+    saveInputValue();
+    setIsEditing(false);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', { completed: todo.completed })}
+    >
+      <label aria-label="Todo status" className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => handleChecker(todo.id)}
+        />
+      </label>
+      {isEditing ? (
+        <form onSubmit={event => event.preventDefault()}>
+          <input
+            ref={inputTodoRef}
+            onBlur={() => handleBlurInput()}
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={queryEditing}
+            onKeyUp={handleClickKeybord}
+            onChange={handleQueryEditChange}
+          />
+        </form>
+      ) : (
+        <>
+          <span
+            data-cy="TodoTitle"
+            onDoubleClick={() => setIsEditing(!isEditing)}
+            className="todo__title"
+          >
+            {todo.title}
+          </span>
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => handleDeleteTodo(todo.id)}
+          >
+            ×
+          </button>
+        </>
+      )}
+
+      <div
+        data-cy="TodoLoader"
+        className={classNames('modal overlay', {
+          'is-active':
+            processingIds.includes(todo.id) || todo.id === 0 || isLoading,
```

Comment:
Let's create separate variable for `processingIds.includes(todo.id) || todo.id === 0 || isLoading`